src/projects/digital/api/circuit/src/internal/assembly/components/gates/BufferAssembler.ts

```diff
@@ -0,0 +1,73 @@
+import {DigitalComponentConfigurationInfo} from "digital/api/circuit/internal/DigitalComponents";
+import {DigitalSim}           from "digital/api/circuit/internal/sim/DigitalSim";
+import {AssemblerParams, AssemblyReason} from "shared/api/circuit/internal/assembly/Assembler";
+import {ComponentAssembler, ComponentBaseShapePrimAssembly} from "shared/api/circuit/internal/assembly/ComponentAssembler";
+import {Schema} from "shared/api/circuit/schema";
+import {V, Vector} from "Vector";
+
+
+export class BufferAssembler extends ComponentAssembler {
+    protected readonly sim: DigitalSim;
+    protected info: DigitalComponentConfigurationInfo;
+
+    public constructor(
+        params: AssemblerParams, sim: DigitalSim, not: boolean
+    ) {
+        super(params, {
+            "outputs": () => ({ origin: V(0.4, 0), target: V(1.2, 0) }),
+            "inputs":  (_, index, total) => {
+                const spacing = 0.5 - this.options.defaultBorderWidth/2;
+                return { origin: V(-0.5, spacing*((total-1)/2 - index)), dir: V(-1, 0) };
+            },
+        }, [
+            // NOT symbol
+            ...((not ? [{
```

Comment:
is using the `GateAssembler` not an option?

Reply:
For some reason I thought it had input port handling hardcoded and I never bothered to double check. GateAssembler works though, switched to that